### ISSUE TITLE
Add batch normalization

### ISSUE THREAD

**Issue by d180cf** [d180cf/dcnn-eval#12]
Oscillations without batch norm:

![image](https://user-images.githubusercontent.com/13122495/32698711-c59b8c58-c75e-11e7-9016-8f4ce680a644.png)

[1] http://ruishu.io/2016/12/27/batchnorm/
[2] https://r2rt.com/implementing-batch-normalization-in-tensorflow.html
[3] https://arxiv.org/pdf/1502.03167.pdf

**Comment by d180cf** [comment]
Without BN:

![image](https://user-images.githubusercontent.com/13122495/32987913-fedffcda-ccac-11e7-8616-afb2d20f23f5.png)

With BN:

![image](https://user-images.githubusercontent.com/13122495/32987917-0cee8a1c-ccad-11e7-88a4-a347e2bc49d3.png)

On the same view:

![image](https://user-images.githubusercontent.com/13122495/32987921-19b789ce-ccad-11e7-9306-bcec182bb94c.png)

Accuracy and loss functions:

![image](https://user-images.githubusercontent.com/13122495/32987927-4ab84518-ccad-11e7-8dbe-ec96f6d1ce4a.png)
